src/types/activity.d.ts

```diff
@@ -42,16 +48,23 @@ type ActivitiesMap = Record<ActivityId, Activity>;
 
 type ActivityDirective = {
   arguments: ArgumentsMap;
+  created_at: string;
   id: ActivityDirectiveId;
+  last_modified_at: string;
+  metadata: ArgumentsMap; // TODO figure out typing of this
```

Comment:
Can we create a new file called `activity-metadata.d.ts` to strongly type this as `ActivityMetadata`? Here are the types:

```ts
type ActivityMetadataValueBoolean = boolean;

type ActivityMetadataValueEnum = number | string;

type ActivityMetadataValueEnumMultiselect = (number | string)[];

type ActivityMetadataValueLongString = string;

type ActivityMetadataValueNumber = number;

type ActivityMetadataValueString = string;

type ActivityMetadataValue =
  | ActivityMetadataValueBoolean
  | ActivityMetadataValueEnum
  | ActivityMetadataValueEnumMultiselect
  | ActivityMetadataValueLongString
  | ActivityMetadataValueNumber
  | ActivityMetadataValueString;

type ActivityMetadataKey = string;

type ActivityMetadata = Record<ActivityMetadataKey, ActivityMetadataValue>;

type ActivityMetadataSchemaBoolean = {
  type: 'boolean';
};

type ActivityMetadataSchemaEnum = {
  enumerates: (number | string)[];
  type: 'enum';
};

type ActivityMetadataSchemaEnumMultiselect = {
  enumerates: (number | string)[];
  type: 'enum_multiselect';
};

type ActivityMetadataSchemaLongString = {
  type: 'long_string';
};

type ActivityMetadataSchemaNumber = {
  type: 'number';
};

type ActivityMetadataSchemaString = {
  type: 'string';
};

type ActivityMetadataSchema =
  | ActivityMetadataSchemaBoolean
  | ActivityMetadataSchemaEnum
  | ActivityMetadataSchemaEnumMultiselect
  | ActivityMetadataSchemaLongString
  | ActivityMetadataSchemaNumber
  | ActivityMetadataSchemaString;
```

Reply:
Sure. Do you want that in this PR or as part of the followup PR?

Reply:
Done.